src/main/java/it/pagopa/transactions/commands/handlers/v1/TransactionSendClosureHandler.java

```diff
@@ -121,14 +121,26 @@ public Mono<Tuple2<Optional<BaseTransactionEvent<?>>, Either<BaseTransactionEven
                             .getTransactionAuthorizationCompletedData();
                     it.pagopa.ecommerce.commons.documents.v1.TransactionActivatedData transactionActivatedData = tx
                             .getTransactionActivatedData();
-                    BigDecimal amount = EuroUtils.euroCentsToEuro(
-                            tx.getPaymentNotices().stream()
-                                    .mapToInt(
-                                            paymentNotice -> paymentNotice.transactionAmount().value()
-                                    ).sum()
+
+                    Integer amount = tx.getPaymentNotices().stream()
+                            .mapToInt(
+                                    paymentNotice -> paymentNotice.transactionAmount().value()
+                            ).sum();
+                    Integer fee = transactionAuthorizationRequestData.getFee();
+                    Integer totalAmount = amount + fee;
+
+                    BigDecimal amountEuroCents = BigDecimal.valueOf(
+                            amount
                     );
-                    BigDecimal fee = EuroUtils.euroCentsToEuro(transactionAuthorizationRequestData.getFee());
-                    BigDecimal totalAmount = amount.add(fee);
+                    BigDecimal feeEuroCents = BigDecimal.valueOf(fee);
+                    BigDecimal totalAmountEuroCents = BigDecimal.valueOf(totalAmount);
+
+                    BigDecimal amountEuro = EuroUtils.euroCentsToEuro(
+                            amount
+                    );
+                    BigDecimal feeEuro = EuroUtils.euroCentsToEuro(fee);
+                    BigDecimal totalAmountEuro = BigDecimal.valueOf(totalAmount);
```

Comment:
```suggestion
                    BigDecimal totalAmountEuro = EuroUtils.euroCentsToEuro(totalAmount);
```
Should this be converted to euro from eurocents?